.github/workflows/windows-pull-request.yaml

```diff
@@ -0,0 +1,79 @@
+name: Windows tests (Pull Request)
+
+on:
+  pull_request:
+    branches:
+      - main
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - labeled
+      - unlabeled
+
+permissions:
+  contents: read
+
+# Limit concurrency by workflow/branch combination.
+#
+# For pull request builds, pushing additional changes to the
+# branch will cancel prior in-progress and pending builds.
+#
+# For builds triggered on a branch push, additional changes
+# will wait for prior builds to complete before starting.
+#
+# https://docs.github.com/en/actions/using-jobs/using-concurrency
+concurrency:
```

Comment:
kinda unfortunate that this has to be a separate workflow, but this seemed like the most straightforward way to achieve the desired behavior